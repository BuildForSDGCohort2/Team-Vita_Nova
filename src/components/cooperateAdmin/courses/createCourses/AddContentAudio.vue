<template>
  <v-container>
    <v-responsive>
      <h3 class="ma-4 " :style="{'font-family': 'IBM Plex Sans'}" style="padding-top: 5%">Untitled Audio</h3>
      <v-card class="text-center justify-center ma-4">
        <v-container  >
          <v-text-field
              v-model="model"
              :items="items"
              :search-input.sync="search"
              hide-selected
              hint="Donâ€™t worry, you can always change this."
              label="Title"
              multiple
              outlined
              persistent-hint
              chips
              color="red"
              class=" title-box "
          >
            <template v-slot:no-data>
              <v-list-item>
                <v-list-item-content>
                  <v-list-item-title>
                    No course matching "<strong>{{ search }}</strong
                  >". Press <kbd>enter</kbd> to create a new one
                  </v-list-item-title>
                </v-list-item-content>
              </v-list-item>
            </template>
          </v-text-field>

          <v-textarea
              name="input-7-1"
              hint="Add an Introduction, detailing what the student would be learning from this module "
              persistent-hint
              color="red"
              outlined
              class=" title-box "
              label="Introduction"
              auto-grow
          ></v-textarea>

          <v-text-field
              color="red"
              outlined
              class=" title-box "
              label="Add Text (Optional)"
              auto-grow
          ></v-text-field>

          <v-text-field
              color="red"
              outlined
              class=" title-box "
              label="Add Links (Optional)"
              auto-grow
          ></v-text-field>

          <v-card-actions>
            <v-btn
                @click="saveCourse"
                :style="{'font-family': 'IBM Plex Sans'}"
                style="background-color:#F8F8F8; color: #FF2E2E; height: 50px !important; width: 120px !important;"
            >Discard</v-btn
            >
            <v-btn
                class="ma-4"
                @click="saveCourse"
                :style="{'font-family': 'IBM Plex Sans'}"
                style="background-color: #FF2E2E; color: #F8F8F8; height: 50px !important; width: 120px !important;"
            >Save</v-btn
            >
          </v-card-actions>
        </v-container>
      </v-card>

      <section :style="{'font-family': 'IBM Plex Sans'}" style="padding-top: 100px; padding-bottom: 100px" class="ma-4">
        <h3>Lesson Settings</h3>
        <v-checkbox v-model="accept" color="red">
          <template v-slot:label>
            Enable discussion on this lesson
          </template>
        </v-checkbox>

        <v-checkbox v-model="agree" color="red">
          <template v-slot:label>
            Make this lesson a free preview
          </template>
        </v-checkbox>
      </section>

      <v-card
          flat
          class="text-center justify-center ma-4"
          max-width="686"
          :style="{'font-family': 'IBM Plex Sans'}"
          style="background: #F8F8F8;border-radius: 11px; height: 60px;"
      >
        <v-container fluid>
          <h6 class="tip-text">
            Tip: Chapter(s) will be set to draft if there are no lessons created
            for it.
          </h6>
        </v-container>
      </v-card>
    </v-responsive>
  </v-container>
</template>

<script>
export default {
  name: "AddContentVideo",

  data: () => ({
    items: ["Software Engineering", "Business Management", "Design Thinking"],
    model: [],
    accept: false,
    agree: false,
    isLoading: false,
    rules: {
      required: v => !!v || "This field is required"
    },
    search: null
  }),

  watch: {
    model(val) {
      if (val.length > 5) {
        this.$nextTick(() => this.model.pop());
      }
    }
  },

  methods: {
    saveCourse() {}
  }
};
</script>

<style scoped></style>
