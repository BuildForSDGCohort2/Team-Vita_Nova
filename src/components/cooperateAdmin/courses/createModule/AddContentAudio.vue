<template>
  <v-responsive>
    <div>
      <v-row>
        <v-col cols="12" md="1"> </v-col>
        <v-col cols="12" md="10">
          <v-row>
            <v-col>
              <h1
                style="font-style: normal;
font-weight: 500;
font-size: 24.0167px;
line-height: 45px;
/* or 188% */


/* lamp-txt-1 */

color: #645262;"
              >
                Untitled Audio
              </h1>
            </v-col>
          </v-row>
          <v-row>
            <v-col>
              <template>
                <v-card class="card mx-auto pa-6" max-width="844" outlined>
                  <ValidationObserver v-slot="{ handleSubmit }">
                    <v-form
                      class="course-form"
                      v-if="!submitted"
                      @submit.prevent="handleSubmit(handleCreateCourseModule)"
                    >
                      <div v-if="errorMsg">
                        <span class="err text-xl-center">{{ errorMsg }}</span>
                      </div>
                      <Loader :loading="loading" :message="message" />

                      <v-row :style="{ 'justify-content': 'center' }">
                        <v-col cols="12" md="8">
                          <ValidationProvider
                            name="Title"
                            rules="required|alpha"
                            v-slot="{ errors }"
                          >
                            <span class="err mt-2 mb-2">{{ errors[0] }}</span>
                            <v-text-field
                              v-model="course.title"
                              class="mt-3"
                              label="Title"
                              outlined
                              color="#FF2E2E"
                              name="courseTitle"
                              chips
                              clearable
                              multiple
                            ></v-text-field>
                          </ValidationProvider>
                        </v-col>
                      </v-row>

                      <v-row :style="{ 'justify-content': 'center' }">
                        <v-col cols="12" md="8" class="ma-auto">
                          <div
                            class="color-title"
                            :style="{ 'font-family': 'IBM Plex Sans' }"
                          >
                            Upload Audio
                          </div>
                          <div v-if="!course.image">
                            <v-btn
                              :style="{ 'font-family': 'IBM Plex Sans' }"
                              block
                              x-large
                              class="#2B1C1C black--text mt-4"
                              @click="$refs.fileInput.click()"
                              dense
                              clearable
                            >
                              <v-icon class="ma-2" color="#645262"
                                >mdi-cloud-upload-outline</v-icon
                              >
                              Choose from files
                            </v-btn>
                            <input
                              ref="fileInput"
                              style="display: none"
                              type="file"
                              accept=".mp3"
                              @change="handleImage"
                            />
                          </div>
                          <div v-else>
                            <v-btn
                              :style="{
                                'font-family': 'IBM Plex Sans',
                                'justify-content': 'start',
                                'text-decoration-line': 'underline'
                              }"
                              block
                              x-large
                              class="#2B1C1C black--text mt-4"
                              @click="$refs.fileInput.click()"
                              dense
                              clearable
                            >
                              {{ image.name }}
                            </v-btn>
                            <input
                              ref="fileInput"
                              style="display: none"
                              type="file"
                              accept=".mp3"
                              @change="handleImage"
                            />
                          </div>
                        </v-col>
                      </v-row>

                      <v-row :style="{ 'justify-content': 'center' }">
                        <v-col cols="12" md="8">
                          <ValidationProvider
                            name="Course Name"
                            rules="required|alpha"
                            v-slot="{ errors }"
                          >
                            <span class="err mt-2 mb-2">{{ errors[0] }}</span>
                            <v-text-field
                              color="red"
                              outlined
                              class=" title-box "
                              label="Add Text (Optional)"
                              auto-grow
                            ></v-text-field>
                          </ValidationProvider>
                        </v-col>
                      </v-row>

                      <v-row :style="{ 'justify-content': 'center' }">
                        <v-col cols="12" md="8">
                          <ValidationProvider
                            name="Course Name"
                            rules="required|alpha"
                            v-slot="{ errors }"
                          >
                            <span class="err mt-2 mb-2">{{ errors[0] }}</span>
                            <v-text-field
                              color="red"
                              outlined
                              class=" title-box "
                              label="Add Text (Optional)"
                              auto-grow
                            ></v-text-field>
                          </ValidationProvider>
                        </v-col>
                      </v-row>

                      <v-row :style="{ 'justify-content': 'center' }">
                        <v-col cols="12" md="8">
                          <v-card-actions>
                            <v-btn
                              @click="saveCourse"
                              :style="{ 'font-family': 'IBM Plex Sans' }"
                              style="background-color:#F8F8F8; color: #FF2E2E; height: 50px !important; width: 120px !important;"
                              >Discard</v-btn
                            >
                            <v-btn
                              class="ma-4"
                              @click="saveCourse"
                              :style="{ 'font-family': 'IBM Plex Sans' }"
                              style="background-color: #FF2E2E; color: #F8F8F8; height: 50px !important; width: 120px !important;"
                              >Save</v-btn
                            >
                          </v-card-actions>
                        </v-col>
                      </v-row>
                    </v-form>
                  </ValidationObserver>
                </v-card>
              </template>

              <section
                :style="{ 'font-family': 'IBM Plex Sans' }"
                style="padding-top: 100px; padding-bottom: 100px"
                class="ma-4"
              >
                <h3>Lesson Settings</h3>
                <v-checkbox v-model="accept" color="red">
                  <template v-slot:label>
                    Enable discussion on this lesson
                  </template>
                </v-checkbox>

                <v-checkbox v-model="agree" color="red">
                  <template v-slot:label>
                    Make this lesson a free preview
                  </template>
                </v-checkbox>
              </section>

              <v-card
                flat
                class="text-center justify-center ma-4"
                max-width="686"
                :style="{ 'font-family': 'IBM Plex Sans' }"
                style="background: #F8F8F8;border-radius: 11px; height: 60px;"
              >
                <v-container fluid>
                  <h6 class="tip-text">
                    Tip: Chapter(s) will be set to draft if there are no lessons
                    created for it.
                  </h6>
                </v-container>
              </v-card>
            </v-col>
          </v-row>
        </v-col>
        <v-col class="mt-5" cols="12" md="1"> </v-col>
      </v-row>
    </div>
  </v-responsive>
</template>

<script>
import { VueEditor } from "vue2-editor";
import Loader from "@/components/ui/loader/Loader";
import { ValidationProvider, ValidationObserver } from "vee-validate";

export default {
  name: "AddContentAudio",
  components: {
    Loader,
    VueEditor,
    ValidationProvider,
    ValidationObserver
  },
  props: {
    value: {
      type: String
    },
    method: { type: Function }
  },
  data: function() {
    return {
      condition: true,
      en: {
        messages: {}
      },
      fields: {
        introduction: {
          max: "Must not be more than 1000 words"
        }
      },
      image: "",
      course: {
        image: null,
        title: "",
        introduction: ""
      },
      style: {
        "font-size": "18px",
        height: "53px",
        background: "#FF2E2E",
        color: "#FFF",
        "font-weight": "bold"
      },
      customToolbar: [
        ["bold", "italic", "underline"],
        [
          { align: "" },
          { align: "center" },
          { align: "right" },
          { align: "justify" }
        ],
        [{ list: "ordered" }, { list: "bullet" }, "link"],
        [{ size: ["small", false, "large", "huge"] }]
      ],
      submitted: false,
      errorMsg: "",
      message: ".",
      loading: false
    };
  },
  computed: {},
  methods: {
    mounted() {
      // Pass a value to the parent through the function
      this.method(this.condition);
    },
    handleImage(e) {
      this.image = e.target.files[0];
      let reader = new FileReader();
      reader.onloadend = () => {
        this.course.image = reader.result;
      };
      reader.readAsDataURL(this.image);
    },
    handleCreateCourseModule() {
      this.loading = true;
      console.log(this.course);
      // const data = { courses: this.course };
    }
  }
};
</script>

<style>
@import "~vue2-editor/dist/vue2-editor.css";

.links {
  /* View All */
  font-weight: normal;
  line-height: 23px;
  letter-spacing: 0.05em;
  text-align: right;
}
.card {
  background: #ffffff;
  border: 2px solid #f8f8f8;
  box-sizing: border-box;
  box-shadow: 0 2px 3px rgba(43, 28, 28, 0.15);
  border-radius: 15px 15px 0 0;
}
.err {
  display: block;
  color: #ff2e2e;
  font-size: 11px;
}
</style>
