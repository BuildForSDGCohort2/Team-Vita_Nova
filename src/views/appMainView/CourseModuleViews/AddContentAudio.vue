<template>
  <v-responsive>
    <AddContentAudio/>
  </v-responsive>
</template>

<script>
import AddContentAudio from "@/components/corporateAdmin/courses/createModule/AddContentAudio";

export default {
  name: "AddContentAudio",
  components: {
    AddContentAudio
  }
};
</script>

<style scoped></style>
