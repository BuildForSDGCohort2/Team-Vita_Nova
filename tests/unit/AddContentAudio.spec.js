import {mount} from "@vue/test-utils";
import AddContentAudio from "@/components/corporateAdmin/courses/createCourses/AddContentAudio";

//testing if the test is properly setup on the run
describe("AddContentAudio", () => {
    const wrapper = mount(AddContentAudio);

    it("setup correctly", () => {
    expect(true).toBe(true);
  });

  //test to check if the title is correct
  it("test page has title", () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.text()).toContain("Untitled Audio");
  });

  it("will change and update the value", () => {
    expect(wrapper.vm.accept).toBe(false);

    // const checkbox = wrapper.vm.$refs.checkbox
    // checkbox.$emit('input', true)
    //
    // expect(wrapper.vm.accept).toBe(true)
  });

  it("sets the correct default data", () => {
    expect(typeof AddContentAudio.data).toBe("function");
    const defaultData = AddContentAudio.data();
    expect(defaultData.search).toBe(null);
  });
});
