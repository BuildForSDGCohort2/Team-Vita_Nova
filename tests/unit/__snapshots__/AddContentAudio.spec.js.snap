// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddContentAudio test page has title 1`] = `
<v-container>
  <v-responsive>
    <h3 class="ma-4 " style="padding-top: 5%;">Untitled Audio</h3>
    <v-card class="text-center justify-center ma-4">
      <v-container>
        <v-text-field items="Software Engineering,Business Management,Design Thinking" hide-selected="" hint="Donâ€™t worry, you can always change this." label="Title" multiple="" outlined="" persistent-hint="" chips="" color="red" class=" title-box "></v-text-field>
        <v-textarea name="input-7-1" hint="Add an Introduction, detailing what the student would be learning from this module " persistent-hint="" color="red" outlined="" label="Introduction" auto-grow="" class=" title-box "></v-textarea>
        <v-text-field color="red" outlined="" label="Add Text (Optional)" auto-grow="" class=" title-box "></v-text-field>
        <v-text-field color="red" outlined="" label="Add Links (Optional)" auto-grow="" class=" title-box "></v-text-field>
        <v-card-actions>
          <v-btn style="background-color: rgb(248, 248, 248); color: rgb(255, 46, 46); height: 50px !important; width: 120px;">Discard</v-btn>
          <v-btn class="ma-4" style="background-color: rgb(255, 46, 46); color: rgb(248, 248, 248); height: 50px !important; width: 120px;">Save</v-btn>
        </v-card-actions>
      </v-container>
    </v-card>
    <section class="ma-4" style="padding-top: 100px; padding-bottom: 100px;">
      <h3>Lesson Settings</h3>
      <v-checkbox color="red"></v-checkbox>
      <v-checkbox color="red"></v-checkbox>
    </section>
    <v-card flat="" max-width="686" class="text-center justify-center ma-4" style="background: rgb(248, 248, 248); border-radius: 11px; height: 60px;">
      <v-container fluid="">
        <h6 class="tip-text">
          Tip: Chapter(s) will be set to draft if there are no lessons created
          for it.
        </h6>
      </v-container>
    </v-card>
  </v-responsive>
</v-container>
`;
